satpy/readers/seviri_l1b_native.py

```diff
@@ -480,6 +516,46 @@ def get_dataset(self, dataset_id, dataset_info):
 
         return dataset
 
+    def pad_hrv_data(self, dataset):
```

Comment:
the areas (row & column numbers) low lower and upper HRV windows differ depending on whether we're dealing with FES or RSS data. this method should probably be divided in two, one method for FES one for RSS so that the implementation could be achieved with stacking different arrays together (instead of item assignment). see comment below.

Reply:
As already discussed I now use dask arrays instead. This in combination with adding also the possibility to pad region of interest data (see Martin's comment), lead quite some differences in the implementation though. There is still only primary function to pad  all data types, but in case of standard FES/RSS data, the relevant lines and columns are read from the trailer, whereas for ROI data this information is read from the secondary header. There are, however, two padding functions in seviri_base now, one for padding in east-west (the function that was/is used in the hrit reader), and another for padding in south-north.